fastlane/Fastfile

```diff
@@ -17,4 +17,69 @@ platform :ios do
       test | xcpretty
       ")
   end
+
+  desc "Runs the test suite and calculates code coverage percentages for each
+  module within the SDK"
+  lane :coverage do
+    MINIMUM_COVERAGE = 100.0
+
+    # Run the test lane to get coverage profdata files
+    unit_tests
+
+    # Ensure the code coverage directory is empty before creating new coverage
+    sh("rm -rf coverage")
+
+    # Use xcov to get the coverage percentages
+    xcov(
+      scheme: "PaymentsCore",
```

Comment:
Why just the `PaymentsCore` scheme here? We have a `UnitTests` scheme if we want to hit all of the module's tests at once. Can we use that instead?

Reply:
Good question! it has to do with how `xccov` (which underpins the `xcov` action) creates coverage data for all targets on build, so running this for each scheme gives us redundant reports (this is how i had it configured initially, and then did some digging to figure out why I was seeing xcov output for all targets three times)

Reply:
You can configure a scheme to only collect partial coverage data for some targets, but since I wanted to analyze all of our targets for coverage data, i thought this was fine

Reply:
🤔  I'm still a little confused. This reads like we're only getting coverage for the `PaymentsCore` module.

Reply:
I imagine that it was getting duplicate reports since our modules have dependencies on each other? Maybe it would be more readable to use the partial coverage syntax, or maybe there is one that lets us specify the entire `PayPal.xcworkspace` proj?

Reply:
Each scheme has this option in under the testing menu: 
<img width="353" alt="image" src="https://user-images.githubusercontent.com/74559931/129928054-a7b7f11d-69f1-459f-9d26-2df47c9fbec8.png">

By running the tests (happens in this file immediately prior to calling `xcov`) we get an `profdata` file for each target in our unit test suite

Since our schemes by default collect coverage data for every target, running `xcov` separately for each target would sort of re-compare every target to the slice of the `profdata` coverage data that xcode generates when running tests.

Maybe a way to make this clearer is to target `Demo` instead - the effect will be the same (all of our targets will be compared to their code coverage data gathered by the test suite)

Reply:
Ah gotcha. Thanks for explaining this 👍